actors/datacap/src/lib.rs

```diff
@@ -439,6 +440,12 @@ impl<T> AsActorResult<T> for Result<T, TokenError> {
     }
 }
 
+impl<T> AsActorResult<T> for Result<T, ReceiverHookError> {
+    fn actor_result(self) -> Result<T, ActorError> {
+        self.map_err(|e| ActorError::unchecked(ExitCode::from(&e), e.to_string()))
```

Comment:
I'm not sure we should do anything about it just now, but there's some chance this `e` carries a system error, which  will then be masked to an illegal actor error or similar if we attempt to exit with it. Maybe file an issue to follow up? Or we could check for system range here now and map to something in user range.